docs/dev/diagrams.md

```diff
@@ -0,0 +1,59 @@
+# Model Diagrams
+
+To aide developers in understanding the various models and how they interoperate for the SSoT project.
+
+```mermaid
+erDiagram
+DiffSyncModel{
+    ClassVar _modelname
+    ClassVar _identifiers
+    ClassVar _shortname
+    ClassVar _attributes
+    ClassVar _children
+    DiffSyncModelFlags model_flags
+    Optional adapter
```

Comment:
Seems weird to have this as just `Optional` - shouldn't it be something like `Optional[Adapter]`?

Same comment on the `ClassVar` above, actually.

Reply:
Not a bad idea, I wasn't too sure on how deep you wanted to get with the typing.